open_schools_platform/student_management/students/services.py

```diff
@@ -67,12 +67,12 @@ def query_handler(query: Query, new_status: str, user: User):
         circle_access = user.has_perm("circles.circle_access", query.recipient)
         student_profile_access = user.has_perm("students.student_profile_access", query.sender)
 
-        if not circle_access:
+        if student_profile_access:
             if query.status != Query.Status.SENT:
                 raise NotAcceptable("Сan no longer change the query")
             if new_status != Query.Status.CANCELED:
                 raise NotAcceptable("User can only set canceled status")
-        elif not student_profile_access:
+        elif circle_access:
             if new_status == Query.Status.CANCELED:
                 raise NotAcceptable("Circle cannot cancel query, it can only decline it")
             if query.status == Query.Status.CANCELED:
```

Comment:
it is not working when student_profile_access and circle_access both true
rework it

Reply:
I think it will work, because if student_profile_access is True, it will not get to the second statement (because of elif) 

I decided that if user has access to query.sender, then he will always act like a parent, not like a member of circle organization.

What do u think?

Reply:
Yeah, I got your idea. But what about accepting this query? Employee that enroll his child can't accept own query?

Reply:
Yes, and I don't know weather it is right or wrong